src/build-defs.cmake

```diff
@@ -5,8 +5,8 @@ endif()
 
 if (UNIX)
   set(CMAKE_C_STANDARD 11)
-  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -L${PG_LIBDIR}")
-  set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -L${PG_LIBDIR}")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -L${PG_LIBDIR} ${PG_LDFLAGS}")
+  set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -L${PG_LIBDIR} ${PG_LDFLAGS}")
```

Comment:
Why only change the `UNIX` flags? This should apply to other platforms too (e.g., `APPLE` and `WIN32`). 

Note, that `UNIX` applies also to Mac/Darwin.

FWIW, this is the output of `pg_config --ldflags` on my Mac:

```
-L/usr/local/opt/openssl/lib/ -Wl,-dead_strips_dylibs
```

The latter part we currently add manually. I think the manual string should be removed in favor of `PG_LDFLAGS`

Reply:
Yes, I agree but since I didn't have access to a Mac I couldn't verify the flags.

In the code there are some more flags, and I am not sure why they are added: `-multiply_defined suppress` , `-Wl,-undefined,dynamic_lookup`, and `-bundle-loader`. Should they be kept?

Also, I have nothing for Windows.

Reply:
Also, I have no idea why `${PG_BINDIR}/postgres` is added. It just seems weird to not use `-lpostgres` and set the path correctly.

Reply:
It seems like we need to keep `-Wl,-undefined,dynamic_lookup` because all symbols in a shared library need to be resolved when building but on [MacOS these options are not set](https://stackoverflow.com/questions/25421479/clang-and-undefined-symbols-when-building-a-library), but out extension have undefined symbols that exists in the host (PostgreSQL binary) and will be resolved when the module is loaded.

Reply:
It also seems like we need to keep the `-multiply_defined,suppress` or get a bunch of warnings about multiply defined symbols.

Reply:
I think it is safe to add `PG_LDFLAGS` to each platform, since that is kind of the purpose of `pg_config` (i.e., to add some platform independence). This should be the baseline. So, I suggest adding `-L${PG_LIBDIR} ${PG_LDFLAGS}` outside any platform-specific tags.

Since `pg_config --ldflags` includes `-Wl,-dead_strips_dylibs` on my mac I think it is safe to remove that part of the custom APPLE flags if we add `PG_LDFLAGS`. Keep the rest for now.

For Windows, nothing needs to change apart from the basic stuff added for all platforms like suggested above.

FWIW, `${PG_BINDIR}/postgres` is an argument to `-bundle_loader`, so that's why it's there. I don't know why it is needed.

Reply:
Moved the adding of `PG_LDFLAGS` to append it after the flags were set. I did not move the `-L${PG_LIBDIR}` since Windows compilers does not use that option format and it works currently.